eth_utils/address.py

```diff
@@ -107,16 +108,18 @@ def to_checksum_address(value: AnyStr) -> ChecksumAddress:
     Makes a checksum address given a supported format.
     """
     norm_address = to_normalized_address(value)
-    address_hash = encode_hex(keccak(text=remove_0x_prefix(norm_address)))
+    address_hash = encode_hex(keccak(text=remove_0x_prefix(HexStr(norm_address))))
```

Comment:
It seems like `to_normalized_address` should return a `HexStr` type... so this wrapping seems fishy to me..

Reply:
It will once https://github.com/ethereum/eth-typing/pull/21 lands.

Reply:
Maybe include a comment linking to that issue/PR here in the code.

Reply:
It turns out I was wrong, thanks to @cburgdorf helping me out with an explanation - but `to_normalized_address()` returns a `HexAddress`, and `remove_0x_prefix()` requires a `HexStr`. So the cast here will be necessary unless we want to relax constraints, i.e...

> remove_0x_prefix(value: Union[HexStr, HexAddress]) -> HexStr:
> add_0x_prefix(value: Union[HexStr, HexAddress]) -> HexStr:

But i'm not sure how I feel about this solution, since any `HexAddress` types will get lost b/c of the logic inside those functions - thoughts?